docs/reference/content/driver-async/tutorials/authentication.md

```diff
@@ -157,9 +157,11 @@ MongoCredential credential = MongoCredential.createMongoX509Credential(user);
 ClusterSettings clusterSettings = ClusterSettings.builder()
                                   .hosts(asList(new ServerAddress("localhost"))).build();
 MongoClientSettings settings = MongoClientSettings.builder()
-                                  .clusterSettings(clusterSettings)
-                                  .credentialList(Arrays.asList(credential))
-                                  .build();
+                .clusterSettings(clusterSettings)
+                .credentialList(Arrays.asList(credential))
+                .streamFactoryFactory(NettyStreamFactoryFactory.builder().build())
```

Comment:
This will cause a nasty leak as is.  Change to:

        EventLoopGroup eventLoopGroup = new NioEventLoopGroup();  // make sure application shuts this down

and then

        NettyStreamFactoryFactory.builder().eventLoopGroup(eventLoopGroup).build();